## 1. Import libraries and project paths

In [1]:
import sys
from pathlib import Path
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

PROJECT_ROOT = Path.cwd().parent if Path.cwd().name == "notebooks" else Path.cwd()
sys.path.append(str(PROJECT_ROOT))

from src.paths import INTERIM_DIR, FIGURES_DIR, OUTPUTS_DIR, create_project_dirs
from src.fia_eda import (
    load_cleaned_dataset,
    create_eda_sample,
    save_dataset_summary,
    create_numeric_summary,
    create_categorical_summary,
    create_state_summary,
    create_species_summary,
    create_forest_type_summary,
    plot_missing_values,
    plot_carbon_distribution,
    plot_log_carbon_distribution,
    plot_dia_vs_carbon,
    plot_ht_vs_carbon,
    plot_cr_vs_carbon,
    plot_carbon_by_state,
    plot_carbon_by_top_species,
    plot_carbon_by_forest_type,
    plot_carbon_by_diameter_class,
    plot_correlation_heatmap
)

create_project_dirs()

print("Project root:", PROJECT_ROOT)
print("Interim folder:", INTERIM_DIR)
print("Figures folder:", FIGURES_DIR)
print("Outputs folder:", OUTPUTS_DIR)

Project root: e:\BA_DV_Project\COMM074_FIA_Project
Interim folder: E:\BA_DV_Project\COMM074_FIA_Project\data\interim
Figures folder: E:\BA_DV_Project\COMM074_FIA_Project\figures
Outputs folder: E:\BA_DV_Project\COMM074_FIA_Project\outputs


## 2. Check cleaned dataset file

In [2]:
cleaned_path = INTERIM_DIR / "fia_tree_carbon_clean_unencoded.parquet"

print("Cleaned dataset exists:", cleaned_path.exists())
print("Cleaned dataset path:", cleaned_path)

Cleaned dataset exists: True
Cleaned dataset path: E:\BA_DV_Project\COMM074_FIA_Project\data\interim\fia_tree_carbon_clean_unencoded.parquet


## 3. Load cleaned dataset

In [3]:
df = load_cleaned_dataset()

print("Cleaned dataset shape:", df.shape)
df.head()

Cleaned dataset shape: (4540549, 55)


,CN,PLT_CN,PREV_TRE_CN,INVYR,STATECD,UNITCD,COUNTYCD,PLOT,SUBP,TREE,...,FLDTYPCD,OWNGRPCD,RESERVCD,SITECLCD,STDAGE,STDSZCD,BALIVE,ALSTK,GSSTK,has_cond_match
0,157825670010854,157531323010854,NaN,1982,1,5,111,90055,104,3,...,NaN,40.0,0.0,5.0,10.0,3.0,NaN,142.921,102.644,True
1,157825671010854,157531323010854,NaN,1982,1,5,111,90055,104,4,...,NaN,40.0,0.0,5.0,10.0,3.0,NaN,142.921,102.644,True
2,157825672010854,157531323010854,NaN,1982,1,5,111,90055,104,5,...,NaN,40.0,0.0,5.0,10.0,3.0,NaN,142.921,102.644,True
3,157825673010854,157531323010854,NaN,1982,1,5,111,90055,104,6,...,NaN,40.0,0.0,5.0,10.0,3.0,NaN,142.921,102.644,True
4,157825692010854,157531325010854,NaN,1982,1,5,111,90057,101,1,...,NaN,40.0,0.0,4.0,30.0,1.0,NaN,121.504,82.939,True


In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 4540549 entries, 0 to 4540548
Data columns (total 55 columns):
 #   Column          Dtype  
---  ------          -----  
 0   CN              int64  
 1   PLT_CN          int64  
 2   PREV_TRE_CN     str    
 3   INVYR           int64  
 4   STATECD         int64  
 5   UNITCD          int64  
 6   COUNTYCD        int64  
 7   PLOT            int64  
 8   SUBP            int64  
 9   TREE            int64  
 10  CONDID          int64  
 11  STATUSCD        int64  
 12  SPCD            int64  
 13  DIA             float64
 14  DIAHTCD         float64
 15  HT              float64
 16  HTCD            str    
 17  ACTUALHT        float64
 18  CR              float64
 19  CCLCD           float64
 20  CARBON_AG       float64
 21  CARBON_BG       float64
 22  DRYBIO_AG       float64
 23  DRYBIO_BG       float64
 24  VOLCFNET        float64
 25  VOLCSNET        float64
 26  VOLBFNET        float64
 27  TPA_UNADJ       float64
 28  PLOT_CN_KEY     int64  

## 4. Create sample for heavy visualisations

In [5]:
eda_sample = create_eda_sample(df, max_rows=100000, random_state=42)

print("Full dataset rows:", len(df))
print("EDA sample rows:", len(eda_sample))

Full dataset rows: 4540549
EDA sample rows: 100000


## 5. Dataset summary

In [6]:
dataset_summary = save_dataset_summary(
    df,
    OUTPUTS_DIR / "eda_dataset_summary.csv"
)

dataset_summary

,row_count,column_count,target_column,number_of_states,number_of_species,min_year,max_year
0,4540549,55,CARBON_AG,6,317,1972,2024


## 6. Numeric variable summary

In [7]:
numeric_summary = create_numeric_summary(
    df,
    OUTPUTS_DIR / "eda_numeric_summary.csv"
)

numeric_summary

,variable,count,mean,std,min,25%,50%,75%,max,missing_count,missing_percentage
0,CN,4540549.0,3.505569e+14,4.178096e+14,1.000000,1.576110e+14,2.054083e+14,3.745489e+14,1.898154e+15,0,0.000000
1,PLT_CN,4540549.0,2.413934e+14,2.900792e+14,3.000000,4.351337e+13,1.575739e+14,2.825015e+14,1.535571e+15,0,0.000000
2,INVYR,4540549.0,2.004361e+03,1.364994e+01,1972.000000,1.997000e+03,2.007000e+03,2.015000e+03,2.024000e+03,0,0.000000
3,STATECD,4540549.0,1.949490e+01,1.610578e+01,1.000000,6.000000e+00,1.300000e+01,2.300000e+01,5.300000e+01,0,0.000000
4,UNITCD,4540549.0,3.521077e+00,2.177414e+00,0.000000,2.000000e+00,3.000000e+00,5.000000e+00,9.000000e+00,0,0.000000
5,COUNTYCD,4540549.0,8.098292e+01,8.274919e+01,1.000000,2.100000e+01,4.700000e+01,1.110000e+02,3.210000e+02,0,0.000000
6,PLOT,4540549.0,3.889357e+04,4.126574e+04,1.000000,5.500000e+01,2.946000e+03,8.944700e+04,9.999700e+04,0,0.000000
7,SUBP,4540549.0,6.174020e+01,1.967164e+02,1.000000,2.000000e+00,3.000000e+00,4.000000e+00,9.990000e+02,0,0.000000
8,TREE,4540549.0,2.232663e+04,2.799743e+05,1.000000,4.000000e+00,1.300000e+01,1.240000e+02,4.879266e+06,0,0.000000
9,CONDID,4540549.0,1.082282e+00,3.042726e-01,1.000000,1.000000e+00,1.000000e+00,1.000000e+00,6.000000e+00,0,0.000000


## 7. Categorical and code variable summary

In [10]:
categorical_summary = create_categorical_summary(
    df,
    OUTPUTS_DIR / "eda_categorical_summary.csv"
)

categorical_summary

,variable,count,missing_count,missing_percentage,n_unique,top_value,top_value_frequency
0,PREV_TRE_CN,1817718,2722831,59.967000,1817718,7189498010497,1
1,HTCD,3370497,1170052,25.768954,6,1.0,2728817
2,ELEV,4401678,138871,3.058463,783,300.0,69848
3,has_plot_match,4540549,0,0.000000,1,True,4540549
4,FLDTYPCD,3469748,1070801,23.583073,154,161.0,526615
5,has_cond_match,4540549,0,0.000000,2,True,4533620


## 8. Missing values after cleaning

In [11]:
missing_rates = df.isna().mean().mul(100).sort_values(ascending=False)

missing_rates.head(30)

VOLBFNET       60.977692
VOLCSNET       60.977692
PREV_TRE_CN    59.967000
ACTUALHT       26.296203
HTCD           25.768954
FLDTYPCD       23.583073
VOLCFNET       23.426793
BALIVE         20.747579
MEASDAY        18.770307
HT             11.506076
CR              9.509401
TPA_UNADJ       4.683310
ELEV            3.058463
GSSTK           3.045469
ALSTK           2.988097
CCLCD           1.944170
RESERVCD        1.905695
STDAGE          1.701975
STDSZCD         0.378897
SITECLCD        0.152603
FORTYPCD        0.152603
OWNGRPCD        0.065256
MEASMON         0.028521
MEASYEAR        0.019315
DRYBIO_AG       0.001828
DRYBIO_BG       0.000022
CARBON_BG       0.000022
STATUSCD        0.000000
SPCD            0.000000
DIA             0.000000
dtype: float64

In [12]:
plot_missing_values(df)

WindowsPath('E:/BA_DV_Project/COMM074_FIA_Project/figures/missing_values.png')

**Interpretation:** The cleaned dataset has no missing values in the target variable required for supervised regression, but several optional predictors still contain missing values. `VOLBFNET`, `VOLCSNET`, and `VOLCFNET` have high missingness and should not be prioritised for modelling, especially because volume-related fields may overlap conceptually with biomass/carbon estimation. `PREV_TRE_CN` is also highly missing and is not relevant for the current single-time tree carbon prediction task. More useful predictors such as `HT`, `CR`, `STDAGE`, `STDSZCD`, `SITECLCD`, `FORTYPCD`, and `OWNGRPCD` have lower missingness and can be handled later through imputation or model-specific preprocessing rather than being removed at this stage.

## 9. Distribution of above-ground carbon storage

In [13]:
df["CARBON_AG"].describe(percentiles=[0.01, 0.05, 0.5, 0.95, 0.99]).to_frame()

,CARBON_AG
count,4.540549e+06
mean,4.948008e+02
std,1.535001e+03
min,6.032000e-03
1%,6.516460e-01
5%,1.288325e+00
50%,9.649500e+01
95%,2.325179e+03
99%,7.391410e+03
max,1.804221e+05


In [14]:
plot_carbon_distribution(df)

WindowsPath('E:/BA_DV_Project/COMM074_FIA_Project/figures/carbon_distribution.png')

**Interpretation:** The raw above-ground carbon distribution is extremely right-skewed. Most trees have relatively low carbon storage, while a small number of very large trees hold exceptionally high carbon values. This pattern is important for the business problem because it suggests that carbon value is not evenly distributed across all trees. A small subset of larger trees may represent a disproportionate share of standing carbon, which is relevant for conservation prioritisation and sustainable harvesting review. From a modelling perspective, this skewness means that errors on high-carbon trees may be especially important and that a log transformation should be considered.

## 10. Log-transformed carbon distribution

In [15]:
df["CARBON_AG_log"] = np.log1p(df["CARBON_AG"])

df["CARBON_AG_log"].describe().to_frame()

,CARBON_AG_log
count,4.540549e+06
mean,4.442790e+00
std,1.997799e+00
min,6.013880e-03
25%,3.409262e+00
50%,4.579801e+00
75%,5.719265e+00
max,1.210306e+01


In [16]:
plot_log_carbon_distribution(df)

WindowsPath('E:/BA_DV_Project/COMM074_FIA_Project/figures/log_carbon_distribution.png')

**Interpretation:** The log-transformed carbon distribution is much more balanced than the raw target distribution, although it still shows multiple peaks. This suggests that the tree population contains several structural groups, such as small trees, medium-sized trees, and larger mature trees. The log transformation reduces the influence of extreme high-carbon trees and may help models such as Linear Regression perform more reliably. However, because the original business problem is about actual above-ground carbon storage, later modelling should clearly explain whether predictions are made on the raw carbon scale or transformed back from the log scale.

## 11. Relationship between diameter and carbon storage

In [17]:
plot_dia_vs_carbon(eda_sample)

WindowsPath('E:/BA_DV_Project/COMM074_FIA_Project/figures/dia_vs_carbon.png')

In [18]:
df[["DIA", "CARBON_AG"]].corr()

,DIA,CARBON_AG
DIA,1.000000,0.827149
CARBON_AG,0.827149,1.000000


**Interpretation:** Diameter at breast height shows a strong positive and clearly non-linear relationship with above-ground carbon storage. Carbon values increase slowly for small trees, then rise much more sharply as diameter increases. This confirms that `DIA` is likely to be one of the most important predictors in the modelling stage. The curved pattern also suggests that purely linear models may underfit larger trees unless engineered features such as `DIA_squared` or non-linear models such as Random Forest and Gradient Boosting are used. In business terms, this supports the idea that larger trees should receive particular attention in carbon-aware harvesting and conservation decisions.

## 12. Relationship between height and carbon storage

In [19]:
plot_ht_vs_carbon(eda_sample)

WindowsPath('E:/BA_DV_Project/COMM074_FIA_Project/figures/ht_vs_carbon.png')

In [20]:
df[["HT", "ACTUALHT", "CARBON_AG"]].corr()

,HT,ACTUALHT,CARBON_AG
HT,1.000000,0.993347,0.706302
ACTUALHT,0.993347,1.000000,0.711568
CARBON_AG,0.706302,0.711568,1.000000


**Interpretation:** Tree height also has a positive relationship with above-ground carbon storage, but the relationship is more dispersed than the diameter-carbon relationship. Taller trees generally store more carbon, but trees of similar height can still vary widely in carbon storage. This suggests that height is useful, but it should be combined with diameter and species information rather than used alone. The correlation heatmap also shows that `HT` and `ACTUALHT` are almost perfectly correlated, so later preprocessing should consider whether both are needed or whether one should be dropped to reduce redundancy.

## 13. Relationship between crown ratio and carbon storage

In [21]:
if "CR" in df.columns:
    plot_cr_vs_carbon(eda_sample)
    display(df[["CR", "CARBON_AG"]].corr())
else:
    print("CR column not available.")

,CR,CARBON_AG
CR,1.000000,0.068727
CARBON_AG,0.068727,1.000000


**Interpretation:** Crown ratio does not show a strong direct relationship with above-ground carbon storage. The scatter plot shows wide variation across most crown ratio values, and the correlation with `CARBON_AG` is weak. This suggests that `CR` is unlikely to be a primary predictor compared with diameter or height. However, it may still capture aspects of tree condition and crown structure, so it can be retained as a secondary predictor if missingness is manageable. Later model interpretation should confirm whether `CR` adds meaningful predictive value.

## 14. Carbon storage by diameter class

In [22]:
df["diameter_class"] = pd.cut(
    df["DIA"],
    bins=[0, 5, 10, 20, 30, 50, 1000],
    labels=["0-5", "5-10", "10-20", "20-30", "30-50", "50+"]
)

diameter_class_summary = (
    df.groupby("diameter_class", observed=True)["CARBON_AG"]
    .agg(["count", "mean", "median"])
    .reset_index()
)

diameter_class_summary

,diameter_class,count,mean,median
0,0-5,1207606,9.782475,4.165030
1,5-10,2044022,113.044586,93.533601
2,10-20,952807,573.822368,474.416191
3,20-30,188167,2285.789347,2189.606706
4,30-50,134901,5998.931111,5412.478056
5,50+,13046,16685.151358,15362.837803


In [23]:
plot_carbon_by_diameter_class(df)

WindowsPath('E:/BA_DV_Project/COMM074_FIA_Project/figures/carbon_by_diameter_class.png')

**Interpretation:** Median carbon storage increases sharply across diameter classes. The largest diameter group (`60+`) has a much higher median carbon value than all smaller groups, confirming that tree size is central to carbon storage. This chart is especially useful for communicating the business value of the model because diameter classes are easier for forest managers to interpret than individual scatter points. The result supports carbon-aware management decisions where larger trees may require closer review before harvesting because their removal could involve a higher carbon opportunity cost.

## 15. Carbon storage by selected state

In [24]:
state_summary = create_state_summary(
    df,
    OUTPUTS_DIR / "eda_state_summary.csv"
)

state_summary

,STATECD,tree_count,mean_carbon,median_carbon,min_carbon,max_carbon
2,13,1459434,238.928741,84.498648,0.200452,66875.541209
0,1,854958,212.945623,79.491161,0.017175,16218.876788
3,23,832606,124.832272,63.498006,0.006032,7742.190219
4,41,629090,1240.720286,200.213116,0.128792,97377.542504
5,53,407671,940.141315,186.453892,0.186841,77959.239492
1,6,356790,1256.139226,196.108805,0.117182,180422.108817


In [25]:
plot_carbon_by_state(df)

WindowsPath('E:/BA_DV_Project/COMM074_FIA_Project/figures/carbon_by_state.png')

**Interpretation:** Median above-ground carbon differs noticeably between the selected states. Oregon (`41`), California (`6`), and Washington (`53`) show higher median carbon values than Alabama (`1`), Georgia (`13`), and Maine (`23`) in this selected dataset. This supports the regional planning angle of the project because carbon storage potential appears to vary geographically. However, these figures should be interpreted as patterns within the selected FIA modelling dataset, not as official state-level FIA population estimates. Later reporting should ideally replace state codes with state names to make the result more accessible to a business audience.

## 16. Carbon storage by species code

In [26]:
species_summary = create_species_summary(
    df,
    OUTPUTS_DIR / "eda_species_summary.csv"
)

species_summary.head(20)

,SPCD,tree_count,mean_carbon,median_carbon,min_carbon,max_carbon
60,131,594859,197.170444,90.549235,0.207023,8027.006938
66,202,378877,2008.862269,422.575060,0.162217,97377.542504
86,316,217297,140.194359,67.527053,0.049317,12947.535307
170,611,212315,151.341979,46.686121,0.241911,10382.810363
1,12,200280,45.482141,28.797305,0.059829,1075.510338
43,111,187008,209.730970,107.958337,0.418884,5905.760131
51,122,132460,1083.917954,261.834121,0.262305,36435.500904
78,263,110919,1028.346492,190.644999,0.240983,57907.166471
243,827,106628,268.774697,85.075916,0.406006,13372.629612
192,694,102550,198.785770,76.022295,0.532472,9000.102349


In [27]:
plot_carbon_by_top_species(df, top_n=15)

WindowsPath('E:/BA_DV_Project/COMM074_FIA_Project/figures/carbon_by_top_species.png')

**Interpretation:** The top species codes show clear differences in median above-ground carbon. Some species codes, especially `202`, `122`, `15`, and `263`, have much higher median carbon than others in the top 15 group. This suggests that species identity may add useful information beyond tree size alone. For the business problem, species-level differences can support restoration planning, conservation prioritisation and forest composition analysis. However, the chart currently uses FIA species codes rather than species names, so the next reporting step should join or map these codes to readable species labels for clearer communication.

## 17. Carbon storage by forest type

In [28]:
forest_type_summary = create_forest_type_summary(
    df,
    OUTPUTS_DIR / "eda_forest_type_summary.csv"
)

forest_type_summary.head(20)

,FORTYPCD,tree_count,mean_carbon,median_carbon,min_carbon,max_carbon
14,161.0,721929,159.678306,73.281769,0.156876,21550.616331
26,201.0,443514,1518.189744,279.939787,0.160114,94161.416306
114,801.0,247417,156.400826,79.098409,0.006032,7742.190219
13,142.0,220542,173.026335,85.314064,0.274637,8398.223849
78,503.0,141475,289.577980,93.478098,0.032432,13178.261184
72,406.0,135723,209.356670,71.382463,0.207023,22922.879306
29,221.0,127529,811.570534,205.877915,0.184941,33721.072876
5,121.0,118849,67.441743,41.150070,0.084872,3781.261190
94,600.0,118815,371.982350,99.932027,0.449916,66875.541209
75,500.0,116853,327.439255,50.828583,0.385226,23729.667626


In [29]:
plot_carbon_by_forest_type(df, top_n=15)

WindowsPath('E:/BA_DV_Project/COMM074_FIA_Project/figures/carbon_by_forest_type.png')

**Interpretation:** Median carbon storage also varies across forest type codes, suggesting that broader ecological context is relevant to tree-level carbon storage. Forest type codes such as `371`, `201`, and `221` show higher median carbon values than several other common forest types. This supports the inclusion of forest type as a categorical predictor in the modelling dataset. As with species, the chart should ideally be translated from FIA forest type codes into readable forest type names before being used in the final report, because management audiences are unlikely to interpret numeric codes directly.

## 18. Correlation between numeric variables

In [30]:
numeric_corr_cols = [
    "CARBON_AG",
    "DIA",
    "HT",
    "ACTUALHT",
    "CR",
    "STDAGE",
    "BALIVE",
    "ALSTK",
    "GSSTK"
]

numeric_corr_cols = [c for c in numeric_corr_cols if c in df.columns]

df[numeric_corr_cols].corr().round(3)

,CARBON_AG,DIA,HT,ACTUALHT,CR,STDAGE,BALIVE,ALSTK,GSSTK
CARBON_AG,1.000,0.827,0.706,0.712,0.069,0.394,0.271,-0.018,0.030
DIA,0.827,1.000,0.860,0.845,0.115,0.438,0.306,-0.071,-0.007
HT,0.706,0.860,1.000,0.993,-0.058,0.333,0.325,0.033,0.088
ACTUALHT,0.712,0.845,0.993,1.000,-0.065,0.330,0.320,0.080,0.140
CR,0.069,0.115,-0.058,-0.065,1.000,0.055,-0.095,-0.118,-0.108
STDAGE,0.394,0.438,0.333,0.330,0.055,1.000,0.385,-0.107,-0.036
BALIVE,0.271,0.306,0.325,0.320,-0.095,0.385,1.000,0.412,0.444
ALSTK,-0.018,-0.071,0.033,0.080,-0.118,-0.107,0.412,1.000,0.854
GSSTK,0.030,-0.007,0.088,0.140,-0.108,-0.036,0.444,0.854,1.000


In [31]:
plot_correlation_heatmap(df)

WindowsPath('E:/BA_DV_Project/COMM074_FIA_Project/figures/correlation_heatmap.png')

**Interpretation:** The correlation heatmap confirms that `DIA` has the strongest numeric relationship with above-ground carbon storage, with a correlation of about 0.83. `HT` and `ACTUALHT` also show strong positive relationships with carbon, both around 0.71, but they are almost perfectly correlated with each other at about 0.99. This creates a potential multicollinearity issue for linear models, so later preprocessing should consider using either `HT` or `ACTUALHT`, rather than both. `CR` has a weak correlation with carbon, while stand and stocking variables show weaker but potentially useful relationships. The negative relationship between `TPA_UNADJ` and tree size/carbon suggests it should be treated carefully and may not be suitable as a main predictor without further interpretation.

## 19. Summary of EDA findings

In [33]:
eda_findings = pd.DataFrame([
    {
        "finding": "Above-ground carbon storage is extremely right-skewed.",
        "business_meaning": "Most trees store relatively low carbon, while a small number of large trees represent high-carbon assets.",
        "modelling_implication": "Consider log transformation, robust metrics, and models that handle skewed targets."
    },
    {
        "finding": "Diameter is the strongest visible and numeric predictor of carbon storage.",
        "business_meaning": "Large-diameter trees are especially important for carbon-aware harvesting and conservation review.",
        "modelling_implication": "Include DIA and engineered features such as DIA_squared and DIA_HT_interaction."
    },
    {
        "finding": "Height is positively related to carbon, but HT and ACTUALHT are almost perfectly correlated.",
        "business_meaning": "Height adds structural information, but duplicate height fields may not add much extra business value.",
        "modelling_implication": "Use either HT or ACTUALHT, or compare both during feature selection to avoid multicollinearity."
    },
    {
        "finding": "Crown ratio has weak direct correlation with carbon storage.",
        "business_meaning": "Crown ratio may describe tree condition, but it is not a primary carbon driver in the EDA.",
        "modelling_implication": "Retain CR as a secondary feature only if missingness and model performance justify it."
    },
    {
        "finding": "Carbon storage differs by state, species code and forest type code.",
        "business_meaning": "Regional and ecological context matters for carbon-aware forest planning.",
        "modelling_implication": "Treat STATECD, SPCD and FORTYPCD as categorical predictors and map codes to readable labels for reporting."
    },
    {
        "finding": "Some optional variables have high missingness.",
        "business_meaning": "Not all FIA fields are equally useful for a scalable decision-support dataset.",
        "modelling_implication": "Drop highly missing or leakage-risk variables such as volume/biomass-related fields from the final modelling feature set."
    }
])

eda_findings_path = OUTPUTS_DIR / "eda_findings_summary.csv"
eda_findings.to_csv(eda_findings_path, index=False)

eda_findings

,finding,business_meaning,modelling_implication
0,Above-ground carbon storage is extremely right...,"Most trees store relatively low carbon, while ...","Consider log transformation, robust metrics, a..."
1,Diameter is the strongest visible and numeric ...,Large-diameter trees are especially important ...,Include DIA and engineered features such as DI...
2,"Height is positively related to carbon, but HT...","Height adds structural information, but duplic...","Use either HT or ACTUALHT, or compare both dur..."
3,Crown ratio has weak direct correlation with c...,"Crown ratio may describe tree condition, but i...",Retain CR as a secondary feature only if missi...
4,"Carbon storage differs by state, species code ...",Regional and ecological context matters for ca...,"Treat STATECD, SPCD and FORTYPCD as categorica..."
5,Some optional variables have high missingness.,Not all FIA fields are equally useful for a sc...,Drop highly missing or leakage-risk variables ...


## 20. Summary

This notebook explored the cleaned FIA tree-level dataset prepared in Notebook 03. The EDA shows that above-ground carbon storage is highly right-skewed, meaning most trees store relatively low carbon while a smaller number of large trees account for much higher carbon values. This supports the business relevance of identifying high-carbon trees when making carbon-aware harvesting, conservation and regional planning decisions.

Tree diameter is the clearest predictor of above-ground carbon storage, with both the scatter plot and correlation heatmap showing a strong positive and non-linear relationship. Height also has a strong positive relationship with carbon, although `HT` and `ACTUALHT` are almost perfectly correlated, meaning the final preprocessing stage should avoid unnecessary duplication. Crown ratio appears much weaker as a direct predictor, but it may still be retained as a secondary condition-related feature if later modelling finds it useful.

The EDA also shows meaningful differences in median carbon storage across selected states, species codes and forest type codes. These patterns support the inclusion of geographic and ecological categorical variables in the modelling dataset. However, the current charts use FIA numeric codes, so the final report should translate key state, species and forest type codes into readable labels where possible.

The next notebook will create the final modelling-ready dataset by selecting predictors, removing leakage-risk variables, engineering features such as `DIA_squared` and `DIA_HT_interaction`, handling remaining missing values through a reproducible preprocessing strategy, and creating train/validation/test splits for the group’s modelling notebooks.